### ISSUE TITLE
kubectl apply deployment in apps group cannot change updataStrategy

### ISSUE THREAD

**Issue by mengqiy** [kubernetes/kubernetes#60340]
This issue https://github.com/kubernetes/kubernetes/issues/24198#issue-148085115 regresses on all versions of Deployment in `apps` group.

The reason is that they don't have the `retainKeys` patchStrategy in their tags.
Deployment in `extensions` group has the `retainKeys` patchStrategy:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/extensions/v1beta1/types.go#L144

We may want to port the `retainKeys` tag for all versions in the apps group.

/kind bug

DaemonSet has similar fields: https://github.com/mengqiy/kubernetes/blob/9b05e2644f55aa531b376d83e1fecf8120151aff/staging/src/k8s.io/api/extensions/v1beta1/types.go#L398-L411
It doesn't give an error if we apply the same steps on it.
Instead, it could become this after change the `updateStrategy` from default to `OnDelete`
```
  updateStrategy:
    rollingUpdate:
      maxUnavailable: 1
    type: OnDelete
```

We may also want to add `retainKeys` to all DeamonSet.

Adding `retainKeys` tag is totally backward compatible.

**Comment by mengqiy** [comment]
cc: @janetkuo

**Comment by mengqiy** [comment]
I believe the tag can be added to deployments with no side-effects.

**Comment by nikhita** [comment]
looks relevant still
/remove-lifecycle rotten

**Comment by mengqiy** [comment]
Fixed by #70436
Closing
